File: backend/src/schemas/pending.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field

from src.models import PendingSection


class PendingFileResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    section: PendingSection
    original_filename: str
    thumbnail_url: str
    scanned_at: datetime


class AssemblePendingRequest(BaseModel):
    section: PendingSection
    title: str | None = Field(default=None, max_length=256)
    file_ids: list[int] = Field(min_length=1)


class SplitWorkRequest(BaseModel):
    title: str | None = Field(default=None, max_length=256)

File: frontend/src/api/pending.ts
import { apiFetch } from "./client";
import type { WorkResponse } from "./works";

export type PendingSection = "book" | "comic";

export interface PendingFileResponse {
  id: number;
  section: PendingSection;
  original_filename: string;
  thumbnail_url: string;
  scanned_at: string;
}

export interface AssemblePendingRequest {
  section: PendingSection;
  title?: string | null;
  file_ids: number[];
}

export function listPending(): Promise<PendingFileResponse[]> {
  return apiFetch<PendingFileResponse[]>("/api/me/pending");
}

export function discardPending(id: number): Promise<void> {
  return apiFetch<void>(`/api/me/pending/${id}`, { method: "DELETE" });
}

export function assemblePending(payload: AssemblePendingRequest): Promise<WorkResponse> {
  return apiFetch<WorkResponse>("/api/me/pending/assemble", { method: "POST", body: payload });
}
